### ISSUE TITLE
Merge Exception: Error when checking model target

### ISSUE THREAD

**Issue by shenglin1987** [fchollet/keras#4318]
Exception: Error when checking model target: expected merge_1 to have 3 dimensions, but got array with shape (3298163, 1)
My code: used to implement cf
vec_dim = 50

uids = np.array([1, 1, 2, 3, 4, 4, 5])
mids = np.array([1, 2, 1, 2, 3, 1, 2])
rates = np.array([3, 4, 5, 1, 3, 4, 5])
num_user = 5
num_POI = 3

userModel = Sequential()
userModel.add(Embedding(num_user, output_dim=vec_dim, input_length=1))
userModel.add(Reshape((vec_dim,)))

poiModel = Sequential()
poiModel.add(Embedding(num_POI, output_dim=vec_dim, input_length=1))
poiModel.add(Reshape((vec_dim,)))

cfModel = Sequential()
merge = Merge([userModel, poiModel], mode=lambda x: np.dot(x[0], x[1]), output_shape=(None,1))
print(merge._output_shape)
cfModel.add(merge)

cfModel.compile(loss='mse', optimizer='sgd')

uids = uids.reshape((len(uids), 1))
mids = mids.reshape((len(mids),1))
rates = rates.reshape((len(rates),1))
cfModel.fit([uids, mids], rates, nb_epoch=10, verbose=2)

I have tried the code in this blog, http://www.fenris.org/2016/03/07/collaborative-filtering-in-keras. But the same problem. Who can help to figure it out?

Please make sure that the boxes below are checked before you submit your issue. Thank you!

- [ y] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [y ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ y] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by shenglin1987** [comment]
This problem is solved. If someone faces the similar case, I give my solution here. 
add this line of code: 
cfModel.add(Reshape((1,))).
Because after Merge(), the outshape is a tuple.
